src/Nethermind/Nethermind.Blockchain/LogTraceDumper.cs

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using System.IO;
+using Nethermind.Core.Crypto;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.Tracing.GethStyle;
+using Nethermind.Evm.Tracing.ParityStyle;
+using Nethermind.Logging;
+using Nethermind.Serialization.Json;
+
+namespace Nethermind.Blockchain
+{
+    public static class BlockTraceDumper
+    {
+        public static void LogDiagnosticTrace(
```

Comment:
Can we make it generic based on BlockTracerBase ? So we don't have to do casts.

Reply:
not part of this change